### PR TITLE
ocpp-v1_6: BootNotificationRequest: Allow optional properties to be set null

### LINKED ISSUE #221: BootNotificationRequest::setChargeBoxSerialNumber(null) throws NullPointerException
I am not sure whether this is the expected behavior, I do not believe it is.
```
new BootNotificationRequest().setChargeBoxSerialNumber(null);
```
throws a `NullPointerException` due to `.length()` in
```
    if (!ModelUtil.validate(chargeBoxSerialNumber, STRING_25_CHAR_MAX_LENGTH)) {
      throw new PropertyConstraintException(
          chargeBoxSerialNumber.length(), validationErrorMessage(STRING_25_CHAR_MAX_LENGTH));
    }
```
being called on a `null` `String`.

Note that `chargeBoxSerialNumber` is optional in `BootNotificationRequest`.

Here's a test for reproduction:
```
package eu.chargetime.ocpp.model.core.test;

import eu.chargetime.ocpp.model.core.BootNotificationRequest;
import org.junit.Before;
import org.junit.Rule;
import org.junit.Test;
import org.junit.rules.ExpectedException;

public class BootNotificationRequestTest {

  @Rule public ExpectedException thrownException = ExpectedException.none();

  private BootNotificationRequest request;

  @Before
  public void setup() {
    request = new BootNotificationRequest();
  }

  @Test()
  public void validate_setChargeBoxSerialNumber_onNull_doesNotThrowNPE() {
    request.setChargeBoxSerialNumber(null);
  }
}
```

### PR DESCRIPTION
**Pull request by bantu**
Fix #221

### CODE REVIEW

File: ocpp-v1_6/src/test/java/eu/chargetime/ocpp/model/core/test/BootNotificationRequestTest.java [hunk 1]
```diff
@@ -0,0 +1,79 @@
+package eu.chargetime.ocpp.model.core.test;
+/*
+   ChargeTime.eu - Java-OCA-OCPP
+
+   MIT License
+
+   Copyright (C) 2016-2018 Thomas Volden <tv@chargetime.eu>
```
Comment (TVolden) [inline ocpp-v1_6/src/test/java/eu/chargetime/ocpp/model/core/test/BootNotificationRequestTest.java:7]: Should probably set a more recent year and maybe add yourself?

**Review by TVolden — COMMENTED** [review]
Looks good. The copy right year should be bumped up to 2022

**Comment by bantu** [comment]
@TVolden Updated

**Review by jmluy — APPROVED** [review]

**Comment by TVolden** [comment]
It's weird. The maven build failed because of some java doc.

**Comment by TVolden** [comment]
Maybe try adding the target source to the pom.xml file

```
--- a/ocpp-v1_6/pom.xml
+++ b/ocpp-v1_6/pom.xml
@@ -116,6 +116,9 @@
             <plugin>
               <groupId>org.apache.maven.plugins</groupId>
               <artifactId>maven-javadoc-plugin</artifactId>
+              <configuration>
+                <source>8</source>
+              </configuration>
               <version>2.9.1</version>
               <executions>
                 <execution>
```

**Comment by bantu** [comment]
@TVolden Added

**Comment by bantu** [comment]
I am able to reproduce the build error using temurim-11. These errors are only warnings when using openjdk-11.

**Comment by bantu** [comment]
Are we seeing this here? https://stackoverflow.com/a/74185633

**Comment by bantu** [comment]
> Are we seeing this here? https://stackoverflow.com/a/74185633

Possible. The version seems to match.

```
 Java configuration:
  Distribution: temurin
  Version: 11.0.17+8
  Path: /opt/hostedtoolcache/Java_Temurin-Hotspot_jdk/11.0.17-8/x64
```

**Comment by TVolden** [comment]
It seems so.

`Command line was: /opt/hostedtoolcache/Java_Temurin-Hotspot_jdk/11.0.17-8/x64/bin/javadoc @options @packages`

**Comment by bantu** [comment]
I've removed my not-working fix commit from this branch.

> So how to fix it?

Can we downgrade to Temurin 11.0.16 or upgrade to a fixed version?

**Comment by bantu** [comment]
Is the javadoc output actually deployed or used somewhere?

**Comment by bantu** [comment]
> Can we downgrade to Temurin 11.0.16 or upgrade to a fixed version?

Added commit doing that.

**Comment by TVolden** [comment]
It worked, amazing work @bantu.

**Pull request merged**